javascript/javascript1/week4/homework6/scripts.js

```diff
@@ -0,0 +1,133 @@
+    return name
+}
+
+//function to capture a task, either to add or remove
+function getTask(command) {
+    if(command.startsWith('Add') || command.startsWith('add')){
+    const endIndex = command.indexOf(' to my todo') // I plan on using slice later, so I'm getting the index at which the slice should stop
+    const task = command.slice(4, endIndex) // get the part of the string between "Add" and "to my todo"
+    return task}
+    //do the same thing except adjust for the words "from" and "Remove"
+    else if (command.startsWith('Remove') || command.startsWith('remove')){
+    const endIndex = command.indexOf(' from my todo') 
+    const task = command.slice(7, endIndex) 
+    return task
+    }
+}
+
+//function to remove a task from the todos
+function removeTask(task) {
+    const taskIndex = todos.findIndex(task);
+    todos.slice(taskIndex, 1)
+    return console.log(`Removed ${task} from your todo`)
+}
+
+// function print date in human redable format
+function getDate(){
+    const today = new Date();
+    const month = today.getMonth(); 
+    const months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+    return console.log(`Today is the ${today.getDate()}. of ${months[month]} ${today.getFullYear()}`)
+}
+
+// function for timer 
+function timer(command) {
+    function getInterval(command){
+        const regex = /(\d+)/ //regular expression to find digits in the command str
+        const match = command.match(regex) // this method will return an array with matching values to the regex 
+        return match ? Number(match[0]) : null; // if true convert the regex to a number and return it
+    }
+    const number = getInterval(command)
+    const interval = number * 60 * 1000;
+    function logDone(){
+        console.log(`Timer Done`)
+    }
+    setTimeout(logDone, interval)
+    console.log(`A timer for ${number} minute(s) has been set`)
+}
+
+
+
+//main function to get answers from the "voice assistant"
+function getReply(command) {
+    // here we deal with the greeting commands
+    if (command.startsWith('Hello my name is')){
+        const person = getName(command)
+        if (!loggedName.includes(person)) {
+            loggedName.push(person)
+            return console.log(`Nice to meet you ${person}`)
+        } else if (loggedName.includes(person)) {
+            return console.log(`Hey ${person}! we met already.`)
+        }
+    }
+    // here we deal with reminding the user their name
+    else if (command.startsWith('What is my name')) {
+        if (loggedName.length === 0) {
+            return console.log(`I can't find a name. \n Please introduce yourself in this way: "Hello my name is" followed by your name`)
```

Comment:
Fantastic error message!

As developers, we are frequently guilty of writing error messages that say things like "Invalid input", or some variation on "I can't do that". This can be frustrating to the user - because from their perspective they have no idea what's actually wrong, or what else they can try.

Your error message instead helps the user to understand how the program does work and gives clues about what they can do about the error. It's much more helpful!